include/exadg/grid/mapping_dof_vector.h

```diff
@@ -478,88 +726,108 @@ initialize_coarse_mappings_from_mapping_dof_vector(
-                                                 constraints_all_levels[h_level - 1]);
+    // create transfer objects
+    dealii::MGLevelObject<dealii::MGTwoLevelTransfer<dim, VectorType>> transfers(0, n_h_levels - 1);
+    for(unsigned int h_level = 1; h_level < n_h_levels; ++h_level)
+    {
+      transfers[h_level].reinit_geometric_transfer(dof_handlers_all_levels[h_level],
+                                                   dof_handlers_all_levels[h_level - 1],
+                                                   constraints_all_levels[h_level],
+                                                   constraints_all_levels[h_level - 1]);
+    }
+
+    // a function that initializes the dof-vector for a given level and dof_handler
+    const std::function<void(unsigned int const, VectorType &)> initialize_dof_vector =
+      [&](unsigned int const h_level, VectorType & vector) {
+        dealii::IndexSet locally_relevant_dofs;
+        dealii::DoFTools::extract_locally_relevant_dofs(dof_handlers_all_levels[h_level],
+                                                        locally_relevant_dofs);
+        vector.reinit(dof_handlers_all_levels[h_level].locally_owned_dofs(),
+                      locally_relevant_dofs,
+                      dof_handlers_all_levels[h_level].get_communicator());
+      };
+
+    dealii::MGTransferGlobalCoarsening<dim, VectorType> mg_transfer_global_coarsening(
+      transfers, initialize_dof_vector);
+
+    // Transfer grid coordinates to coarser h-levels.
+    // The dealii::DoFHandler object will not be used for global coarsening.
+    dealii::DoFHandler<dim>           dof_handler_dummy;
+    dealii::MGLevelObject<VectorType> grid_coordinates_all_levels(0, n_h_levels - 1);
+    mg_transfer_global_coarsening.interpolate_to_mg(dof_handler_dummy,
+                                                    grid_coordinates_all_levels,
+                                                    grid_coordinates_fine_level);
+
+    // initialize mapping for all coarse h-levels using the dof-vectors with grid coordinates
+    AssertThrow(coarse_mappings.size() == n_h_levels - 1,
+                dealii::ExcMessage(
+                  "coarse_mappings does not have correct size relative to coarse_triangulations."));
+
+    for(unsigned int h_level = 0; h_level < coarse_mappings.size(); ++h_level)
+    {
+      coarse_mappings[h_level] =
+        std::make_shared<MappingDoFVector<dim, Number>>(*coarse_triangulations[h_level]);
+
+      // grid_coordinates_all_levels describes absolute coordinates -> use an uninitialized mapping
+      // in order to interpret the grid coordinates vector as absolute coordinates and not as
+      // displacements.
+      std::shared_ptr<dealii::Mapping<dim> const> mapping_dummy;
+      coarse_mappings[h_level]->initialize_mapping_from_dof_vector(
+        mapping_dummy, grid_coordinates_all_levels[h_level], dof_handlers_all_levels[h_level]);
```

Comment:
@kronbichler `dof_handler_all_levels` is a local variable in this function. This will cause the code to terminate in the simplex case (because of `MappingFEField` using smart pointers to the `DoFHandler`) when using these mapping objects inside the multigrid preconditioner, where the code crashes when initializing the MatrixFree objects for all the multigrid levels with the error message 

```bash
The violated condition was: 
    pointed_to_object_is_alive
Additional information: 
    The object pointed to is not valid anymore.
```
I could change the interfaces in ExaDG towards `std::shared_ptr` and store such a shared pointer to the dof-handler in the class `MappingDoFVector`. Since these requires some changes I want to ask beforehand what you would suggest to solve this problem?  (Regarding dealii, I could imagine it might be a better software design (on the dealii-side) to use arguments of type `std::shared_ptr<DoFHandler>` in the constructor of  `dealii::MappingFEField`. This would in my opinion more clearly communicate to the user that the class `MappingFEField` still needs the constructor arguments after construction of the object.)

Reply:
Yes, I would also suggest to keep the `DoFHandler` in `MappingDoFVector`, as discussed in person.